AdobeStockImage/Model/GetRelatedImages.php

```diff
@@ -116,8 +120,11 @@ private function serializeRelatedImages(array $images): array
                     }
                     $itemData[$attribute->getAttributeCode()] = $attribute->getValue();
                 }
-                $data[] = $itemData;
+                if ($image->getId() !== $imageId) {
+                    $data[] = $itemData;
+                }
             }
+            $data = (count($data) > $limit-1 ? array_pop($data) : $data);
```

Comment:
```suggestion
            $data = array_slice($data, 0, $limit);
```

Reply:
@sivaschenko .. I'm slightly disagree with this suggested changes because it will remove the first image regardless of the preview image is available in the related images or not and user/merchant might miss the first image. Correct me if I'm wrong
Thanks

Reply:
if the original image was removed before this line of code - nothing will be sliced as the array length will be equal to limit. Correct me if I'm wrong

Reply:
Yes, But as we have incremented the limit to get an extra image which we need to handle if preview image is not available in the related image so the length of the array will be greater than the limit is defined in the javascript and it will display an extra image in the related image section.